### PR TITLE
Refactor events/llmp.rs

### PR DESCRIPTION
**Pull request by tokatoka**
separate these into multiple files.
use builder pattern
fix sampling_rate

### CODE REVIEW

**Comment by tokatoka** [comment]
i''ll rewrite most of these constructors in builder pattern.

**Comment by domenukk** [comment]
I think instead of a sampling rate we should do something like a `throttle` method for min ms between `fire`s

**Comment by domenukk** [comment]
That way if it finds something new later in the process, other nodes will also get it. A sample method wouldn't do the same.

File: libafl/src/events/mod.rs [hunk 1]
```diff
@@ -621,6 +624,10 @@ impl<S> EventFirer for NopEventManager<S>
 where
     S: State,
 {
+    fn sample(&self, _corpus_count: usize) -> bool {
```
Comment (domenukk) [inline libafl/src/events/mod.rs:627]: As noted earlier, let's do a `throttle` function instead (or even call it `should_skip`/`should_send` to be flexible)

Comment (tokatoka) [reply libafl/src/events/mod.rs]: DONE

File: fuzzers/nautilus_sync/src/lib.rs [hunk 2]
```diff
@@ -120,13 +120,14 @@ pub extern "C" fn libafl_main() {
...
 
     let mut event_converter = opt.bytes_broker_port.map(|port| {
-        LlmpEventConverter::on_port(
-            shmem_provider.clone(),
-            port,
-            Some(NautilusToBytesInputConverter::new(&context)),
-            none_input_converter!(),
-        )
-        .unwrap()
+        LlmpEventConverterBuilder::new()
```
Comment (domenukk) [inline fuzzers/nautilus_sync/src/lib.rs:123]: Launcher uses `LlmpEventConverter::builder()`

Comment (domenukk) [reply fuzzers/nautilus_sync/src/lib.rs:123]: `typed_builder` in general uses `::builder()`, we may want to keep this consistent (?)

File: libafl/src/events/mod.rs (hunk 1 above)
Comment (domenukk) [reply libafl/src/events/mod.rs]: Awsome

File: libafl/src/events/tcp.rs [hunk 3]
```diff
@@ -711,6 +689,14 @@ where
     EMH: EventManagerHooksTuple<S>,
     S: State,
 {
+    fn should_send(&self) -> bool {
+        if let Some(throttle) = self.throttle {
+            libafl_bolts::current_time() - self.last_sent > throttle
```
Comment (domenukk) [inline libafl/src/events/tcp.rs:694]: Looks like `self.last_sent` is never set(?)

File: fuzzers/nautilus_sync/src/lib.rs (hunk 2 above)
Comment (domenukk) [reply fuzzers/nautilus_sync/src/lib.rs:123]: I guess `Builder::new()` is the pattern for non-typed-builder builders.
We could add a `::builder` method to the base struct anyway in this case

File: libafl/src/events/tcp.rs (hunk 3 above)
Comment (tokatoka) [reply libafl/src/events/tcp.rs:694]: nice catch 😉

File: fuzzers/nautilus_sync/src/lib.rs (hunk 2 above)
Comment (tokatoka) [reply fuzzers/nautilus_sync/src/lib.rs:123]: in this way?
```
fn builder() -> Self {
 Self::new()
}
```

Comment (tokatoka) [reply fuzzers/nautilus_sync/src/lib.rs:123]: maybe in other pr. now all libafl builder use new(), and not builder(), for its constructor

Comment (domenukk) [reply fuzzers/nautilus_sync/src/lib.rs:123]: They don't - all typed_builders use `::builder()` (and there's quite a few of them)

**Pull request merged**